### PR TITLE
have app loggers log to their parent logger

### PR DESCRIPTION
**Pull request by devoncarew**
- have app loggers log to their parent logger (instead of directly to stdout). This does the right thing wrt the verbosity level (it respects `-v`)

@tvolkert

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 1]
```diff
@@ -769,19 +769,17 @@ class AppInstance {
 
 /// A [Logger] which sends log messages to a listening daemon client.
 class _AppRunLogger extends Logger {
-  _AppRunLogger(this.domain, this.app, { this.logToStdout: false });
+  _AppRunLogger(this.domain, this.app, { this.logToParent });
 
   AppDomain domain;
   final AppInstance app;
-  final bool logToStdout;
+  final Logger logToParent;
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: maybe just `parent`?

Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: +1

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 2]
```diff
@@ -769,19 +769,17 @@ class AppInstance {
 
 /// A [Logger] which sends log messages to a listening daemon client.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: I don't really understand this class. What is the logToParent option for? It seems to make the entire class just act like the provided parent, in which case, why bother creating the object at all?

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: `logToParent` changes the behavior to using the parent to handle all stdout. This is valuable for IntelliJ's implementation of the flutter run -v option - previously we weren't passing trace level messages to the IDEs. An equivalent change would be to keep the current behavior - not add logToParent - but to pass all message levels, including trace, to IDEs.

> why bother creating the object at all?

With or without those changes, this class sends progress messages to the IDE, so it can use that info to provide nice, integrated progress UIs.

File: packages/flutter_tools/flutter_tools.iml [hunk 3]
```diff
@@ -30,16 +34,10 @@
...
-      <excludeFolder url="file://$MODULE_DIR$/test/replay/osx/simulator_application_binary/platform/packages" />
-      <excludeFolder url="file://$MODULE_DIR$/test/replay/osx/simulator_application_binary/process/packages" />
-      <excludeFolder url="file://$MODULE_DIR$/test/replay/osx/simulator_application_binary/vmservice/packages" />
       <excludeFolder url="file://$MODULE_DIR$/test/replay/packages" />
-      <excludeFolder url="file://$MODULE_DIR$/test/src/base/packages" />
-      <excludeFolder url="file://$MODULE_DIR$/test/src/ios/packages" />
+      <excludeFolder url="file://$MODULE_DIR$/test/runner/packages" />
       <excludeFolder url="file://$MODULE_DIR$/test/src/packages" />
       <excludeFolder url="file://$MODULE_DIR$/tool/packages" />
     </content>
```
Comment (Hixie) [inline packages/flutter_tools/flutter_tools.iml]: Are these changes expected?

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: Would it make sense to have two subclasses of _AppRunLogger, one for when you want to send all events to the IDE, and one for when you only want to send progress events to the IDE but want status, error, and trace events to instead be logged to the console?

I still don't really understand what this is for.

The actual change (from logToStdout to having an explicit parent) seems like an improvement, I'm just stuck on the underlying code that you're changing here.

File: packages/flutter_tools/flutter_tools.iml (hunk 3 above)
Comment (devoncarew) [inline packages/flutter_tools/flutter_tools.iml]: They were, so I'd have less scm churn here. Note that the not-yet-released version of IntelliJ won't generate these excludes, so there won't be as many changes to these files.

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 2 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: Re: two loggers, that's totally doable - and I do agree that the current state is a bit confusing.

The main value this class adds is letting IDEs convert the spinner progress info available in the CLI to progress mechanisms that make more sense in the IDE (indeterminate progress bars). And what I'm trying to do w/ this change is to plumb more of the verbose info through, so people can use the equivalent of `flutter run -v` to help diagnose when there are issues launching apps.

> The actual change (from logToStdout to having an explicit parent) seems like an improvement

lgtm then, or should I work on specialized _AppRunLogger subclasses?

Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: You should at a minimum leave a TODO mentioning that it might make sense to refactor this code.

File: packages/flutter_tools/flutter_tools.iml (hunk 3 above)
Comment (Hixie) [inline packages/flutter_tools/flutter_tools.iml]: Can we do this in a separate PR maybe? I don't really understand why this particular file exists in our repo at all, but that's probably a discussion for another bug.

**Comment by Hixie** [comment]
LGTM but @tvolkert is the right person to review this.

**Comment by devoncarew** [comment]
Sounds good (and added a TODO).

**Comment by tvolkert** [comment]
LGTM modulo @Hixie's comment about `flutter_tools.iml`

**Comment by devoncarew** [comment]
Sure, reverted the .iml file change.
